### PR TITLE
PMM-6014 Prepare Release Candidate Job

### PR DESCRIPTION
**Pull request by puneet0191**
This job basically does the following:

1) Prepares a release branch on pmm-submodules repo (ex. `release-2.17.0`) with updated .gitmodules using release repo for  each submodule mentioned in the list `pmm_submodules` submodule and `pmm-2.x.y` for dependent submodules as shown in `dependent_submodules` 
2) It rewinds submodules for the newly created Release branch and triggers autobuild jobs 
3) In case we want to rebuild with new changes for release, the job automatically skips the creation of branches and runs rewind and auto-build jobs. 

implementation is based on how we prepare release-2.16 branch for 2.16.0 release, see here: https://github.com/Percona-Lab/pmm-submodules/blob/release-2.16/.gitmodules

### CODE REVIEW

**Comment by denisok** [comment]
so, if we have done 2.17.0 but there is a need for 2.17.1 but 2.18 is still in progress.
We would create branch for 2.17.1 from release-2.17.0 ? Running with parameters VERSION=2.17.1 GIT_BRANCH=release-2.17.0 ?

**Comment by puneet0191** [comment]
@denisok yes, you are right, we just need to pass the release branch in the case of Hotfix

**Review by denisok — APPROVED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]

**Review by atymchuk — APPROVED** [review]

**Review by BupycHuk — COMMENTED** [review]
Do we have a job to merge all created branches back or delete them?

File: pmm/pmm2-prepare-release-candidate.groovy [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        "qan-api2",
+        "pmm-update",
+        "pmm-server",
+        "grafana-dashboards",
+        "pmm-api-tests",
+        "pmm-ui-tests",
+        "pmm-qa"
+    ]
+    def dependent_submodules = [
+        "mysqld_exporter",
```
[the remark this answers is not in the document]
Comment (puneet0191) [reply pmm/pmm2-prepare-release-candidate.groovy:56]: okay will do

**Comment by puneet0191** [comment]
> Do we have a job to merge all created branches back or delete them?
@BupycHuk no, do you mean merging release-2.x.y branches to main/master?

File: pmm/pmm2-prepare-release-candidate.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (puneet0191) [reply pmm/pmm2-prepare-release-candidate.groovy:56]: I have added, could you please check again @BupycHuk

**Review by denisok — APPROVED** [review]

**Pull request merged**